lib/foreignKeyValidationFactory.js

```diff
@@ -1,30 +1,30 @@
-var traverse = require('traverse');
-var async = require('async');
+const traverse = require('traverse');
+const async = require('async');
 
 module.exports = function foreignKeyValidationFactory(foreignKeys) {
 	return foreignKeyValidation;
 
 	function foreignKeyValidation(document, options, callback) {
-		var valuesToCheck = getValuesToCheck(foreignKeys, document);
+		const valuesToCheck = getValuesToCheck(foreignKeys, document);
 		if (Object.keys(valuesToCheck).length === 0) {
 			return callback(null, []);
 		}
-		var tasks = Object.keys(valuesToCheck).map(function (propertyName) {
-			var foreignKeyCheckFunction = foreignKeys[propertyName];
+		const tasks = Object.keys(valuesToCheck).map(function (propertyName) {
+			const foreignKeyCheckFunction = foreignKeys[propertyName];
 			return getForeignKeyCheckTask(foreignKeyCheckFunction, valuesToCheck[propertyName], options);
 		});
 		async.parallel(tasks, function (err, results) {
 			if (err) {
 				return callback(err);
 			}
-			var errors = Array.prototype.concat.apply([], results);
+			const errors = Array.prototype.concat.apply([], results);
 			return callback(null, errors);
 		});
 	}
 };
 
 function getValuesToCheck(foreignKeys, document) {
-	var memo = {};
+	let memo = {};
```

Comment:
Do we actually redefine memo or just add properties?

```javascript
memo.foo = 'foo';
```

does not mean we redefined memo and it can be `const` in this case

Reply:
We do just in the first line of the collapsed code part 😄 
<img width="542" alt="image" src="https://github.com/debitoor/schemagic/assets/100693724/70ae176d-35da-4351-8794-ea2596efc72e">